### PR TITLE
Build images

### LINKED ISSUE #11: Package up official fluvio connectors
When running `fluvio connector start --config ./config.yaml`, this contains a set of connectors. Some of these connectors will be done in this repo.

Right now, we have a `syslog` connector and a [`test-connector` connector](https://github.com/infinyon/fluvio-connectors/tree/main/test-connector). Part of CI for this project will be [`make smoke-test`](https://github.com/infinyon/fluvio-connectors/blob/5f7f9c73e21d293ad41235ce23f02dea127ccb57/Makefile#L1-L2) to run the `test-connector`.

In order to do this, we either need to package up the connector as a docker container or publish it/all connectors before hand. Which of these two is open for discussion. I think it would help with development if we had it able to be done on the fly at `fluvio connector start`.

### PR DESCRIPTION
**Pull request by tjtelan**
Resolves #11 
 
* Makefile target to support syslog and test connector
* Add wrapper Dockerfile
* Adding container build into CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,3 +43,100 @@ jobs:
...
+          path: container-build 
+
+      #- name: Install system toolchain - aarch64-unknown-linux-musl
+      #  if: ${{ matrix.rust-target == 'aarch64-unknown-linux-musl' }}
+      #  run: |
+      #    apt update
+      #    apt install -y gcc-aarch64-linux-gnu 
+
+      - name: Build containers
+        run: make IMAGE_NAME=${{ matrix.image-name }} CONNECTOR_NAME=${{ matrix.connector-name }} official-containers
```
Comment (simlay) [inline .github/workflows/ci.yml:141]: Oh, you fancy. Parallel build.

File: Makefile [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+TARGET_FLAG=$(if $(TARGET),--target $(TARGET),)
+
+# Connectors
+TEST_CONNECTOR_BIN=$(if $(TARGET),./target/$(TARGET)/$(BUILD_PROFILE)/test-connector,./target/$(BUILD_PROFILE)/test-connector)
+SYSLOG_BIN=$(if $(TARGET),./target/$(TARGET)/$(BUILD_PROFILE)/fluvio-syslog,./target/$(BUILD_PROFILE)/fluvio-syslog)
+
+#CONNECTOR_NAME?=test-connector
+#IMAGE_NAME?=infinyon/fluvio-connect-test-connector
+CONNECTOR_NAME?=
+IMAGE_NAME?=
```
Comment (simlay) [inline Makefile]: If these parameters aren't specified to `make`, what's the behavior?

**Comment by tjtelan** [comment]
I think there's more to do, especially w/r/t supporting arm64, but I think most of the foundation is ready. We may want to revisit the workflow if we need a lot more flexibility for adding connectors w/o touching CI

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tjtelan) [reply .github/workflows/ci.yml:141]: Yeah! But in development you can just run `make official-containers` and you'll build the test-connector by default

File: container-build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ENV CONNECTOR_NAME=${CONNECTOR_NAME}
+
+# Copy the connector somewhere the image's $PATH
+COPY ${CONNECTOR_NAME} /usr/local/bin/${CONNECTOR_NAME}
+RUN chmod +x /usr/local/bin/${CONNECTOR_NAME}
+
+# The startup script uses this file to get the connector binary name
+RUN echo ${CONNECTOR_NAME} | tee /CONNECTOR_NAME
+
+ENTRYPOINT [ "sh", "-c", "tini $CONNECTOR_NAME" ]
```
Comment (tjtelan) [inline container-build/Dockerfile:17]: Proactively, I will say I added `tini` bc I was testing running a container locally, and it wouldn't respect `Ctrl+c`. I had to wait for the connector to time out w/o `tini`

**Review by simlay — APPROVED** [review]
Looks good! Merge this when you're ready.

File: Makefile (hunk 2 above)
Comment (simlay) [reply Makefile]: You already answered my question.

**Comment by tjtelan** [comment]
Welp, I guess bors isn't going to connect at the moment. Will merge now and try bors again later.

**Pull request merged**